modules/core_api/fsw/inc/cfe_resourceid.h

```diff
@@ -0,0 +1,219 @@
+
+/**
+ * \brief Determine if two derived (app-specific) IDs are equal
+ *
+ * This generic routine is implemented as a macro so it is agnostic to the actual argument type,
+ * and it will evaluate correctly so long as the argument type is based on the CFE_RESOURCEID_BASE_TYPE.
+ */
+#define CFE_RESOURCEID_TEST_EQUAL(id1, id2) CFE_ResourceId_Equal(CFE_RESOURCEID_UNWRAP(id1), CFE_RESOURCEID_UNWRAP(id2))
+
+/**
+ * @brief Convert a resource ID to an integer.
+ *
+ * This is primarily intended for logging purposes, such was writing
+ * to debug console, event messages, or log files, using printf-like APIs.
+ *
+ * For compatibility with C library APIs, this returns an "unsigned long"
+ * type and should be used with the "%lx" format specifier in a printf
+ * format string.
+ *
+ * @note No assumptions should be made about the actual integer value,
+ * such as its base/range.  It may be printed, but should not be modified
+ * or tested/compared using other arithmetic ops, and should never be used
+ * as the index to an array or table.  See the related function
+ * CFE_ResourceId_ToIndex() for cases where a zero-based array/table index
+ * is needed.
+ *
+ * @sa CFE_ResourceId_FromInteger()
+ *
+ * @param[in]   id    Resource ID to convert
+ * @returns Integer value corresponding to ID
+ */
+static inline unsigned long CFE_ResourceId_ToInteger(CFE_ResourceId_t id)
+{
+    return ((unsigned long)CFE_RESOURCEID_UNWRAP(id));
+}
+
+/**
+ * @brief Convert an integer to a resource ID.
+ *
+ * This is the inverse of CFE_ResourceId_ToInteger(), and reconstitutes
+ * the original CFE_ResourceId_t value from the integer representation.
+ *
+ * This may be used, for instance, where an ID value is parsed from a text
+ * file or message using C library APIs such as scanf() or strtoul().
+ *
+ * @sa CFE_ResourceId_ToInteger()
+ *
+ * @param[in]   Value    Integer value to convert
+ * @returns ID value corresponding to integer
+ */
+static inline CFE_ResourceId_t CFE_ResourceId_FromInteger(unsigned long Value)
+{
+    return ((CFE_ResourceId_t)CFE_RESOURCEID_WRAP(Value));
+}
+
+/**
+ * @brief Compare two Resource ID values for equality
+ *
+ * @param[in]   id1    Resource ID to check
+ * @param[in]   id2    Resource ID to check
+ * @returns true if id1 and id2 are equal, false otherwise.
+ */
+static inline bool CFE_ResourceId_Equal(CFE_ResourceId_t id1, CFE_ResourceId_t id2)
+{
+    return (CFE_RESOURCEID_UNWRAP(id1) == CFE_RESOURCEID_UNWRAP(id2));
+}
+
+/**
+ * @brief Check if a resource ID value is defined
+ *
+ * The constant #CFE_RESOURCEID_UNDEFINED represents an undefined ID value,
+ * such that the expression:
+ *
+ *      CFE_ResourceId_IsDefined(CFE_RESOURCEID_UNDEFINED)
+ *
+ * Always returns false.
+ *
+ * @param[in]   id    Resource ID to check
+ * @returns True if the ID may refer to a defined entity, false if invalid/undefined.
+ */
+static inline bool CFE_ResourceId_IsDefined(CFE_ResourceId_t id)
+{
+    return (CFE_RESOURCEID_UNWRAP(id) != 0);
```

Comment:
The value to denote whether a resource ID is undefined should be used here instead of just a hard 0. This should be cfe_resourceid_api_typedefs.h:CFE_RESOURCEID_UNDEFINED.

Reply:
Can use the macro

Reply:
Dee https://github.com/nasa/cFE/issues/1302